### PR TITLE
#2315 Program Dashboard Valid / Invalid Donor Filter

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

Adds ElasticSearch handler for Valid/Invalid Donor filters

Relates to: https://github.com/icgc-argo/platform-ui/pull/2420

**Type of Change**

- [ ] Bug
- [x] New Feature

### CODE REVIEW

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts [hunk 1]
```diff
@@ -452,6 +452,23 @@ const programDonorSummaryEntriesAndStatsResolver: (esClient: Client) => DonorEnt
 				const dnaTNMatchedPairQuery = esb.boolQuery().should(shouldQueries);
 				queries.push(dnaTNMatchedPairQuery);
 			}
+
+			if (field === EsDonorDocumentField.validWithCurrentDictionary && filter.values.length > 0) {
+				for (const value of filter.values) {
+					if (value === 'VALID' || value === 'INVALID') {
```
Comment (samrichca) [inline src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts:458]: please use enum (see `tumorNormalMatchedPairStatus.TUMOR_NORMAL_MATCHED_PAIR`)

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts [hunk 2]
```diff
@@ -452,6 +452,23 @@ const programDonorSummaryEntriesAndStatsResolver: (esClient: Client) => DonorEnt
...
+						queries.push(
+							esb
+								.boolQuery()
+								.must(
+									esb.termsQuery(
+										EsDonorDocumentField.validWithCurrentDictionary,
+										value === 'VALID' ? true : false,
```
Comment (samrichca) [inline src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts:465]:
```suggestion
										value === 'VALID',
```

the ternary is redundant, a comparison will return the boolean value you want

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts (hunk 1 above)
Comment (demariadaniel) [reply src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts:460]: Yes, updated!

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts (hunk 2 above)
Comment (demariadaniel) [reply src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts]: Classic bit of overthinking/not thinking there hahaha, thanks updated

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts [hunk 3]
```diff
@@ -452,6 +453,24 @@ const programDonorSummaryEntriesAndStatsResolver: (esClient: Client) => DonorEnt
 				const dnaTNMatchedPairQuery = esb.boolQuery().should(shouldQueries);
 				queries.push(dnaTNMatchedPairQuery);
 			}
+
+			if (field === EsDonorDocumentField.validWithCurrentDictionary && filter.values.length > 0) {
+				const shouldQueries = [];
```
Comment (samrichca) [inline src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts:458]:
```suggestion
				const shouldQueries: Query[] = [];
```

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts [hunk 4]
```diff
@@ -452,6 +453,24 @@ const programDonorSummaryEntriesAndStatsResolver: (esClient: Client) => DonorEnt
...
+				for (const value of filter.values) {
+					if (value === 'VALID' || value === 'INVALID') {
```
Comment (samrichca) [inline src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts:460]:
```suggestion
					if (value === validWithCurrentDictionaryStatus.VALID || value === validWithCurrentDictionaryStatus.INVALID) {
```

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts (hunk 3 above)
Comment (demariadaniel) [reply src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts]: Updated

File: src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts (hunk 4 above)
Comment (demariadaniel) [reply src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts]: Updated

**Pull request merged**